src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -36,9 +36,10 @@ public Movie add(Movie movie) {
 
     @Override
     public Optional<Movie> get(Long id) {
-        Session session = HibernateUtil.getSessionFactory().openSession();
-        Optional<Movie> getResult = Optional.ofNullable(session.get(Movie.class, id));
-        session.close();
-        return getResult;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(Movie.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find movie №" + id, e);
+        }

```

Comment:
its not a mistake, but FYI: u won't get exception if there is no entity with this id